Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -558,21 +807,102 @@ class Gx2Fitter {
             "a hole.");
 
+        // Transport the covariance to the surface
+        stepper.transportCovarianceToBound(state.stepping, *surface,
+                                           freeToBoundCorrection);
+
+        // TODO generalize the update of the currentTrackIndex
+        auto& fittedStates = *result.fittedStates;
+
+        // Add a <trackStateMask> TrackState entry multi trajectory. This
+        // allocates storage for all components, which we will set later.
+        typename traj_t::TrackStateProxy trackStateProxy =
+            fittedStates.makeTrackState(Gx2fConstants::trackStateMask,
+                                        result.lastTrackIndex);
+        const std::size_t currentTrackIndex = trackStateProxy.index();
+
+        // Set the trackStateProxy components with the state from the ongoing
+        // propagation
+        {
+          trackStateProxy.setReferenceSurface(surface->getSharedPtr());
+          // Bind the transported state to the current surface
+          auto res = stepper.boundState(state.stepping, *surface, false,
+                                        freeToBoundCorrection);
+          if (!res.ok()) {
+            result.result = res.error();
+            return;
+          }
+          // Not const since, we might need to update with scattering angles
+          auto& [boundParams, jacobian, pathLength] = *res;
+
+          // For material surfaces, we also update the angles with the
+          // available scattering information
+          // We can skip the if here, since we already know, that we do
+          // multipleScattering and have material
+          {
+            ACTS_DEBUG("    Update parameters with scattering angles.");
+            const auto scatteringMapId = scatteringMap->find(geoId);
+            ACTS_VERBOSE("    scatteringAngles:\n"
+                         << scatteringMapId->second.getScatteringAngles()
+                         << "\n    boundParams before the update:\n"
+                         << boundParams);
+            boundParams.parameters() +=
+                scatteringMapId->second.getScatteringAngles();
+            ACTS_VERBOSE("    boundParams after the update:\n" << boundParams);
+          }
+
+          // Fill the track state
+          trackStateProxy.predicted() = boundParams.parameters();
+          trackStateProxy.predictedCovariance() = state.stepping.cov;
+
+          trackStateProxy.jacobian() = jacobian;
+          trackStateProxy.pathLength() = pathLength;
+
+          {
+            stepper.update(state.stepping,
+                           transformBoundToFreeParameters(
+                               trackStateProxy.referenceSurface(),
+                               state.geoContext, trackStateProxy.predicted()),
+                           trackStateProxy.predicted(),
+                           trackStateProxy.predictedCovariance(), *surface);
+          }
```

Comment:
nitpick: block for function call looks unnecessary

Reply:
I was thinking about this a bit. We have the same scheme before with
```
if (doMaterial) {
...
}
```
so I wanted to keep it the same. But now I think it might be overall more confusing than beneficial^^